client/src/modules/stock/exit/exit.js

```diff
@@ -117,16 +126,57 @@ function StockExitController(
         headerCellFilter : 'translate',
         cellTemplate : 'modules/stock/exit/templates/expiration.tmpl.html',
       },
-      { field : 'actions', width : 25, cellTemplate : 'modules/stock/exit/templates/actions.tmpl.html' },
+      {
+        field : 'actions',
+        width : 25,
+        cellTemplate : 'modules/stock/exit/templates/actions.tmpl.html',
+      },
     ],
     data : vm.stockForm.store.data,
     fastWatch : true,
     flatEntityAccess : true,
+    showGridFooter : true,
+    gridFooterTemplate,
+    onRegisterApi,
   };
 
   // exposing the grid options to the view
   vm.gridOptions = gridOptions;
 
+  /**
+   * @method exportGrid
+   * @description export the content of the grid to csv.
+   * this function is helpful for handling the export of lot row which are in the grid to a csv file
+   * in a correct format.
+   */
+  vm.exportGrid = () => {
+    const columns = vm.gridOptions.columnDefs
+      .filter(col => col.displayName && col.displayName.length)
+      .map(col => ({ displayName : $translate.instant(col.displayName), width : col.width }));
+
+    const rows = vm.gridOptions.data.map(row => {
+      const code = row.inventory && row.inventory.code ? row.inventory.code : null;
```

Comment:
The user can add rows in the grid without defines inventories, in this case the grid is populate but it doesn't have the object that we need (`inventory`), the user is free to export in any time.

Reply:
So, instead of having a long list of boolean checks, can we just do something like the [the PatientInvoiceItemService](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/invoices/patientInvoiceItem.service.js#L39) where we set a variable to track whether the form should have data or not?  Something like:
```js
  if (row.initialized) {
     // export this row, assume all values are there, throw errors if any values are missing
  }  else {
     // skip this row or refuse to export or something else 
  }
```

It just makes it easier to read instead of a ton of `&& ? :` logic.